src/super_gradients/training/models/segmentation_models/unet/unet_decoder.py

```diff
@@ -0,0 +1,182 @@
+                 min_decoder_channels: int = 1,
+                 **up_block_kwargs):
+        """
+
+        :param skip_channels_list: num_channels list of skip feature maps from the encoder.
+        :param up_block_repeat_list: `num_repeats` arg list to be passed to the UpFuseBlocks.
+        :param skip_expansion: skip expansion ratio value, before fusing the skip features from the encoder with the
+            decoder features, a projection convolution is applied upon the encoder features to project the num_channels
+            by skip_expansion as follows: `num_channels = skip_channels * skip_expansion
+        :param decoder_scale: num_channels width ratio between encoder stages and decoder stages.
+        :param min_decoder_channels: The minimum num_channels of decoder stages. Useful i.e if we want to keep the width
+            above the num of classes. The num_channels of a decoder stage is determined as follows:
+                `decoder_channels = max(encoder_channels * decoder_scale, min_decoder_channels)`
+        :param up_block_types: list of UpBlockType.
+        :param is_skip_list: List of flags whether to use feature-map from encoder stage as skip connection or not. Used
+            to not apply projection convolutions if a certain encoder feature is not aggregate with the decoder.
+        :param up_block_kwargs: init parameters for fuse blocks.
+        """
+        super().__init__()
+        up_block_types = [
+            UpBlockType.from_block_id(block_type) if isinstance(block_type, int) else block_type
+            for block_type in up_block_types
+        ]
+        # num_channels list after encoder features projections.
+        self.up_channels_list = [max(int(ch * decoder_scale), min_decoder_channels) for ch in skip_channels_list]
+        # Reverse order to up-bottom order, i.e [stage4_ch, stage3_ch, ... , stage1_ch]
+        self.up_channels_list.reverse()
+        # Remove last stage num_channels, as it is the input to the decoder.
+        self.up_channels_list.pop(0)
+
+        is_skip_list.reverse()
+        is_skip_list += [False]
+
+        self.projection_blocks, skip_channels_list = self.make_skip_projection(skip_channels_list, skip_expansion,
+                                                                               is_skip_list, min_decoder_channels)
+        skip_channels_list = skip_channels_list.copy()
+        skip_channels_list.reverse()
+
+        self.up_stages = nn.ModuleList()
+        in_channels = skip_channels_list.pop(0)
+        skip_channels_list.append(None)
+        for i in range(len(up_block_types)):
+            self.up_stages.append(
+                up_block_types[i].up_block_cls(in_channels, skip_channels_list[i], self.up_channels_list[i],
+                                               num_repeats=up_block_repeat_list[i], **up_block_kwargs))
+            in_channels = self.up_channels_list[i]
+
+    def make_skip_projection(self,
```

Comment:
please make it private (_)

Reply:
Done [0a56285](https://github.com/Deci-AI/super-gradients/pull/401/commits/0a56285e7b3123ad1b0c377e28f4589398942969)